packages/sentry/src/index.ts

```diff
@@ -0,0 +1,57 @@
+import type * as SentryApi from '@sentry/react';
+
+type SupportedMethods =
+	| 'addBreadcrumb'
+	| 'captureEvent'
+	| 'captureException'
+	| 'captureMessage'
+	| 'configureScope'
+	| 'withScope';
+interface QueueDataMethod< Method extends SupportedMethods > {
+	f: Method;
+	a: Parameters< typeof SentryApi[ Method ] >;
+}
+
+export interface A8cSentryGlobals {
+	a8cSentryCallQueue?: Array< Readonly< QueueDataMethod< SupportedMethods > > >;
+	a8cDispatchSentryMethodCall?: < Method extends SupportedMethods >(
+		method: Method,
+		args: Parameters< typeof SentryApi[ Method ] >
+	) => void;
+}
+
+declare global {
+	// eslint-disable-next-line @typescript-eslint/no-empty-interface
+	interface Window extends A8cSentryGlobals {}
+}
+
+function dispatchSentryMethodCall< Method extends SupportedMethods >(
+	method: Method,
+	args: Parameters< typeof SentryApi[ Method ] >
+) {
+	if ( window.a8cDispatchSentryMethodCall ) {
+		window.a8cDispatchSentryMethodCall( method, args );
+	} else {
+		window.a8cSentryCallQueue = window.a8cSentryCallQueue || [];
+		window.a8cSentryCallQueue.push( { f: method, a: args } );
```

Comment:
I think this will grow the call queue indefinitely if Sentry is never loaded in an environment.

Reply:
The reason this wasn't an issue in the other file is that we clear the call queue there if Sentry doesn't get loaded (which works when that file is also in charge of loading Sentry).

Reply:
Ah yep, you're right. If `status` was `'error'` or `'disabled'` we'd stop pushing items onto the array. I should be able to fix that with yet another global 😆 

Although maybe I should have a go using Gutenberg filters instead like you mentioned above.

Reply:
That said, if there is no sentry environment at all -- like if `@automattic/components` was used in a different application -- those statuses wouldn't get populated